Content.Server/IgnitionSource/IgnitionSourceSystem.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System.Diagnostics;
+using System.Linq;
+using Content.Server.Atmos.EntitySystems;
+using Content.Server.Light.Components;
+using Content.Shared.Light.Component;
+using Robust.Server.GameObjects;
+using Serilog;
+
+namespace Content.Server.IgnitionSource;
+
+/// <summary>
+/// This handles ignition
+/// </summary>
+public sealed class IgnitionSourceSystem : EntitySystem
+{
+    /// <inheritdoc/>
+    ///
+
+    [Dependency] private readonly AtmosphereSystem _atmosphereSystem = default!;
+    [Dependency] private readonly TransformSystem _transformSystem = default!;
+
+
+    public override void Initialize()
+    {
+        // ? does this need to exist
+    }
+
+
+    public void SetState(EntityUid uid, IgnitionSourceComponent component, bool newState)
+    {
+        component.State = newState;
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+
+        foreach (var (component,transform) in EntityQuery<IgnitionSourceComponent,TransformComponent>())
+        {
+            var source = component.Owner;
+
+            if(EntityManager.TryGetComponent(source, out ExpendableLightComponent? expendable))
```

Comment:
You may want to use events instead of setting a boolean on the component. This also means you won't need to loop over every ignition source ever update